src/cattr/gen.py

```diff
@@ -181,6 +183,17 @@ def make_dict_structure_fn(cl: Type, converter, **kwargs):
                 f"    res['{ian}'] = {struct_handler_name}(o['{kn}'], __c_t_{an})"
             )
     lines.append("    }")
+    if forbid_extra_fields:
+        post_lines.append("  allowed_fields = {")
+        post_lines += [f"    '{a.name}'," for a in attrs]
+        post_lines += [
+            "  }",
+            "  unknown_fields = set(o.keys()) - allowed_fields",
+            "  if unknown_fields:",
```

Comment:
I would condense this into a single line
```suggestion
            "  if o.keys() - allowed_fields:",
```

Reply:
Except then I can't re-use it to list the unknown fields in the exception message.  Or would you prefer to recalculate it again?

Reply:
Er good point!